vault/external_tests/identity/aliases_test.go

```diff
@@ -250,3 +255,881 @@ func TestIdentityStore_RenameAlias_CannotMergeEntity(t *testing.T) {
+		"canonical_id":   entityIdAlice,
+		"mount_accessor": mountAccessor,
+	})
+	if err != nil {
+		t.Fatalf("err:%v resp:%#v", err, aliasResp)
+	}
+
+	aliasIdAlice := aliasResp.Data["id"].(string)
+	if aliasIdAlice == "" {
+		t.Fatal("Alias ID not present in response")
+	}
+
+	// Now create a new unrelated entity and alias, clara-smith
+	entityRespClara, err := client.Logical().Write("identity/entity", map[string]interface{}{
+		"name": "clara-smith",
+	})
+	if err != nil {
+		t.Fatalf("err:%v resp:%#v", err, entityRespAlice)
+	}
+	if entityRespClara == nil {
+		t.Fatalf("expected a non-nil response")
+	}
+	entityIdClara := entityRespClara.Data["id"].(string)
+	if entityIdClara == "" {
+		t.Fatal("EntityID not present in response")
+	}
+
+	aliasResp, err = client.Logical().Write("identity/entity-alias", map[string]interface{}{
+		"name":           "clara",
+		"canonical_id":   entityIdClara,
+		"mount_accessor": mountAccessorGitHub,
+	})
+	if err != nil {
+		t.Fatalf("err:%v resp:%#v", err, aliasResp)
+	}
+
+	aliasIdClara := aliasResp.Data["id"].(string)
+	if aliasIdClara == "" {
+		t.Fatal("Alias ID not present in response")
+	}
+
+	// Perform entity merge
+	mergeResp, err := client.Logical().Write("identity/entity/merge", map[string]interface{}{
+		"to_entity_id":    entityIdBob,
+		"from_entity_ids": []string{entityIdAlice, entityIdClara},
+	})
+	if err == nil {
+		t.Fatalf("Expected error upon merge. Resp:%#v", mergeResp)
+	}
+	if !strings.Contains(err.Error(), "conflicting alias mount accessors between toEntity and fromEntity") {
+		t.Fatalf("Error was not due to conflicting alias mount accessors. Error: %v", err)
+	}
+	if !strings.Contains(err.Error(), entityIdAlice) {
+		t.Fatalf("Did not identify alice's entity (%s) as conflicting. Error: %v", entityIdAlice, err)
+	}
+	if !strings.Contains(err.Error(), entityIdBob) {
+		t.Fatalf("Did not identify bob's entity (%s) as conflicting. Error: %v", entityIdBob, err)
+	}
+	if !strings.Contains(err.Error(), entityIdClara) {
+		t.Fatalf("Did not identify clara's alias (%s) as conflicting. Error: %v", entityIdClara, err)
+	}
+	if !strings.Contains(err.Error(), aliasIdAlice) {
+		t.Fatalf("Did not identify alice's alias (%s) as conflicting. Error: %v", aliasIdAlice, err)
+	}
+	if !strings.Contains(err.Error(), aliasIdBob) {
+		t.Fatalf("Did not identify bob's alias (%s) as conflicting. Error: %v", aliasIdBob, err)
+	}
+	if !strings.Contains(err.Error(), aliasIdClara) {
+		t.Fatalf("Did not identify bob's alias (%s) as conflicting. Error: %v", aliasIdClara, err)
+	}
+	if !strings.Contains(err.Error(), mountAccessor) {
+		t.Fatalf("Did not identify mount accessor %s as being reason for conflict. Error: %v", mountAccessor, err)
+	}
+	if !strings.Contains(err.Error(), mountAccessorGitHub) {
+		t.Fatalf("Did not identify mount accessor %s as being reason for conflict. Error: %v", mountAccessorGitHub, err)
+	}
+}
+
+func TestIdentityStore_MergeEntities_SameMountAccessor_ThenUseAlias(t *testing.T) {
```

Comment:
Consider using the helper function createEntityAndAlias to reduce verbosity.

Reply:
Thanks, good suggestion. I'll apply this to all of the tests.

Reply:
Thanks, much nicer.